kernel/comps/virtio/src/systree.rs

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Virtio node registration to systree
+
+extern crate alloc;
+
+use alloc::{borrow::Cow, collections::BTreeMap, string::ToString, sync::Arc, vec::Vec};
+
+use ostd::mm::{VmReader, VmWriter};
+use spin::RwLock;
+use systree::{
+    Result, RootNode, SysAttrSet, SysBranchNode, SysNode, SysNodeId, SysNodeType, SysObj,
+};
+
+/// Virtio root node under systree
+#[derive(Debug)]
+pub struct VirtioNode {
+    id: SysNodeId,
+    name: alloc::string::String,
+    attrs: SysAttrSet,
+    children: RwLock<BTreeMap<alloc::string::String, Arc<dyn SysObj>>>,
+}
+
+impl VirtioNode {
+    pub fn new() -> Arc<Self> {
+        Arc::new(Self {
+            id: SysNodeId::new(),
+            name: "virtio".to_string(),
```

Comment:
Define the name as `SysStr` and initialize it with `Cow::from("virtio")`. This avoid unnecessary heap allocation. Use the (statically) borrowed form of `SysStr` as much as possible in all locations.